libevmasm/EVMAssemblyStack.h

```diff
@@ -82,6 +90,7 @@ class EVMAssemblyStack: public AbstractAssemblyStack
 private:
 	langutil::EVMVersion m_evmVersion;
 	std::optional<uint8_t> m_eofVersion;
+	Assembly::OptimiserSettings m_optimiserSettings;
```

Comment:
`Assembly::OptimiserSettings` contains the EVM version, which makes storing `m_evmVersion` separately a bit redundant.

Or, alternatively, we could take it out of `Assembly::OptimiserSettings`. At the Solidity level we don't consider it a part of optimizer settings because it has a broader application (even though it's *a* setting the optimizer takes).

Reply:
I think we should remove it from the optimiser settings. Both evm version and eof version are args of the assembly constructor so passing it there in terms of optimiser settings is already redundant as well. It would also create a separation between settings that switch optimizer components on and off and other configurations.

Reply:
Good. I see that we apparently didn't even have any sanity checks that these two versions were consistent so it was possible to assemble with one and then optimize with another.